src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -31,7 +35,32 @@
     private static final boolean DEBUG_STATS = false;
 
     /**
-     * Loads the alignment regions and sequence of all locally-assembled contigs from the text file they are in;
+     * Converts a {@link GATKRead}, particularly the alignment and sequence information in it to the custom {@link AlignmentRegion} format.
+     */
+    @VisibleForTesting
+    static Tuple2<Iterable<AlignmentRegion>, byte[]> convertToAlignmentRegions(final Iterable<GATKRead> reads) {
```

Comment:
done with 2 pass streaming, one is essentially for cautiously checking that there's a primary alignment in the input.